seatunnel-connectors/seatunnel-connector-flink-socket/src/main/java/org/apache/seatunnel/flink/connector/socket/source/SocketSourceFunction.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * <p>It splits records by the given byte delimiter (`\n` by default) and delegates the decoding to
+ * a pluggable {@link DeserializationSchema}.
+ *
+ * <p>Note: This is only an example and should not be used in production. The source function is not
+ * fault-tolerant and can only work with a parallelism of 1.
+ */
+public final class SocketSourceFunction extends RichSourceFunction<RowData>
+        implements ResultTypeQueryable<RowData> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SocketSourceFunction.class);
+
+    private final String hostname;
+    private final int port;
+    private final byte byteDelimiter;
+    private final DeserializationSchema<RowData> deserializer;
+    private static final int WAIT_TIME = 1000;
+
+    private volatile boolean isRunning = true;
+    private Socket currentSocket;
+
+    public SocketSourceFunction(
+            String hostname,
+            int port,
+            byte byteDelimiter,
+            DeserializationSchema<RowData> deserializer) {
+        this.hostname = hostname;
+        this.port = port;
+        this.byteDelimiter = byteDelimiter;
+        this.deserializer = deserializer;
+    }
+
+    @Override
+    public TypeInformation<RowData> getProducedType() {
+        return deserializer.getProducedType();
+    }
+
+    @Override
+    public void open(Configuration parameters) throws Exception {
+        deserializer.open(
+                RuntimeContextInitializationContextAdapters.deserializationAdapter(
+                        getRuntimeContext()));
+    }
+
+    @Override
+    public void run(SourceContext<RowData> ctx) throws Exception {
+        while (isRunning) {
+            // open and consume from socket
+            try (final Socket socket = new Socket()) {
+                currentSocket = socket;
+                socket.connect(new InetSocketAddress(hostname, port), 0);
+                try (InputStream stream = socket.getInputStream()) {
+                    ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+                    int b;
+                    while ((b = stream.read()) >= 0) {
+                        // buffer until delimiter
+                        if (b != byteDelimiter) {
+                            buffer.write(b);
+                        }
+                        // decode and emit record
+                        else {
+                            ctx.collect(deserializer.deserialize(buffer.toByteArray()));
+                            buffer.reset();
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                LOG.warn("Meet exception when reading from socket", e);
+            }
+            Thread.sleep(WAIT_TIME);
+        }
+    }
+
+    @Override
+    public void cancel() {
+        isRunning = false;
+        try {
+            currentSocket.close();
+        } catch (Exception t) {
+            // ignore
```

Comment:
Normally we rename the `t` as `ignored` instead of adding comment here

Reply:
Make sense, thanks.